### PR TITLE
fix: display name validation (client)

### LINKED ISSUE #22578: [Bug]: Input field validation is missing for display name
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

There is no input field validation for display name. (Note: Add validations in backend as well)
image

### Steps To Reproduce

1.  Got to profile page from top right menu
2. Edit the display name

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by eco-monk**
## Description
Added client validation for display name field.
- Allows all language characters, `.`, `-`, `'` and space. 
(Every other special character will throw an error e.g. html content)
- Save field when pressing enter / on blur.

Fixes #22578 

Media
https://www.loom.com/share/ad96ed687c0740379326462ffc05a0ec

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by hetunandu — DISMISSED** [review]

**Review by akash-codemonk — DISMISSED** [review]

File: app/client/src/pages/UserProfile/General.tsx [hunk 1]
```diff
@@ -33,8 +34,31 @@ const ForgotPassword = styled.a`
...
+  message: string;
+} => {
+  const notEmpty = notEmptyValidator(value);
+  if (!notEmpty.isValid) {
+    return notEmpty;
+  }
+  if (!new RegExp(`^[${ALL_LANGUAGE_CHARACTERS_REGEX} 0-9.'-]+$`).test(value)) {
+    return {
+      isValid: false,
+      message: "No special characters allowed except .'-",
```
Comment (mohanarpit) [inline app/client/src/pages/UserProfile/General.tsx:50]: Nit: Please move all strings to `messages.ts` file.

File: app/client/src/pages/UserProfile/General.tsx [hunk 2]
```diff
@@ -79,11 +103,17 @@ function General() {
...
               fill={false}
-              onChange={onNameChange}
+              onBlur={saveName}
+              onChange={setName}
+              onKeyPress={(ev: React.KeyboardEvent) => {
+                if (ev.key === "Enter") {
+                  saveName();
+                }
+              }}
               placeholder="Display name"
```
Comment (mohanarpit) [inline app/client/src/pages/UserProfile/General.tsx:115]: Please move this as well to `messages.ts` file

Comment (eco-monk) [reply app/client/src/pages/UserProfile/General.tsx]: Moved all labels to `messages.ts`
image

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**